### ISSUE TITLE
Switch: add non-checkbox version

### ISSUE THREAD

**Issue by ianmcburnie** [eBay/skin#833]
Slightly controversial this one. When we first created the switch (a long time ago) we just saw it as a different visual treatment of a checkbox. So a checkbox was used under the hood.

Over the past few years one distinguishing feature of switch has emerged: it does not post form data. It executes JavaScript. To rephrase: toggling a checkbox should not have an instant effect on the page until the user submits their selection. Whereas toggling a switch *should* have an instant effect on the page.

Bearing this in mind, I think we might want to drop the input element in favour of just using divs/spans + ARIA. The controversial aspect is that some might say we lose a graceful fallback to a checkbox in the case of JavaScript failing. But as the very purpose of the switch is to execute JavaScript, this fallback now seems unnecessary.

Also see: #832
